roomba/roomba.py

```diff
@@ -275,6 +279,10 @@ def on_message(self, mosq, obj, msg):
             self.decode_topics(self.master_state)  # publish all values
             self.time = time.time()
 
+        # call the callback functions
+        for func in self.on_message_callbacks:
+            func(msg)
```

Comment:
you cant return msg in callback, this is raw data which came from irobot
you need to use `json_data`